### PR TITLE
fix(pdb): skip reading rows of invalid pages

### LINKED ISSUE #72: rekordcrate-pdb - Error: no variants matched at 0x307f
out

### PR DESCRIPTION
**Pull request by Swiftb0y**
as per doc of `pdb::Page::page_flags` certain
pages contain garbage data and thus should
not try to read rowdata.

fixes #72

### CODE REVIEW

**Comment by Swiftb0y** [comment]
This solution just skips the rows but still reads the page (which then contains no rows). Alternatively one could check if the page has no data and then not add it to vector of pages. That solution would be a little more complicated though I fear.

File: src/pdb/mod.rs [hunk 1]
```diff
@@ -181,6 +181,17 @@ impl Header {
...
+#[binrw]
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+struct PageFlags(u8);
+
+impl PageFlags {
+    #[must_use]
+    pub fn page_has_data(&self) -> bool {
+        (self.0 & 0x40) == 0
+    }
+}
```
[the remark this answers is not in the document]
Comment (Swiftb0y) [reply src/pdb/mod.rs:193]: Didn't know that, sure. I just don't know when I'll get around to it.

**Comment by Swiftb0y** [comment]
Couldn't get `modular_bitfield` to play nice with clippy. Seems like the modular_bitfield codegen must be updated. `modular_bitfield` has not been updated in almost two years...

**Comment by Holzhaus** [comment]
Hmm, I think we are using it because it was [suggested by the `binrw` docs](https://docs.rs/binrw/latest/binrw/docs/attribute/index.html#using-map-on-a-struct-to-create-a-bit-field). Maybe we need to switch?  Maybe the [`bitflags` crate](https://docs.rs/bitflags/latest/bitflags/) can be used?

If you restore the original PR, I can merge it now to fix the bug and we just put further improvements on the TODO list a.k.a. the issue tracker.

**Comment by Swiftb0y** [comment]
restored

**Comment by Swiftb0y** [comment]
Thanks

**Comment by Holzhaus** [comment]
Thanks for fixing.

**Pull request merged**